Patch origin: upstream
Patch status: unclear; patch attached to bug

https://bugs.php.net/bug.php?id=53829

--- php-5.3.17/ext/zlib/zlib.c
+++ php-5.3.17/ext/zlib/zlib.c
@@ -58,6 +58,18 @@
 # endif
 #endif
 
+/*
+ * zlib include files can define the following preprocessor defines which rename
+ * the corresponding PHP functions to gzopen64, gzseek64 and gztell64 and thereby
+ * breaking some software, most notably PEAR's Archive_Tar, which halts execution
+ * without error message on gzip compressed archivesa.
+ *
+ * This only seems to happen on 32bit systems with large file support.
+ */
+#undef gzopen
+#undef gzseek
+#undef gztell
+
 #if defined(HAVE_UNISTD_H) && defined(PHP_WIN32)
 # undef HAVE_UNISTD_H
 #endif
